wrappers.c

```diff
@@ -417,7 +419,7 @@ struct unixdirent * __unix_readdir(DIR *dirp)
     return NULL;
 }
 
-sighandler_t __unix_signal(int signum, sighandler_t handler)
```

Comment:
It seems like FreeBSD does have `sig_t` though, I think we should add something like this:

```c
#ifdef __FreeBSD__ 
typedef sig_t sighandler_t;
#endif
```

Reply:
I don't think that it's a good idea to use GNU as a baseline and then add an exception for each system that's not GNU. Maintaining many platform-specific code paths with `#ifdefs` on our end could get out of hand quickly.

 Rather we should leverage the fact that GNU libc knows about POSIX, and knows about BSD, and offers so-called Feature Test Macros for compatibility purposes. There's a good writeup about FTMs on on LWN: https://lwn.net/Articles/590381/ and more info in the [`feature_test_macros(7)` man page](`feature_test_macros(7)` man page).

Concretely using -D_DEFAULT_SOURCE instead of -D_GNU_SOURCE would make BSD `sig_t` translate into GNU `sighandler_t`. Howeve I'm interested in the reasoning behind using _GNU_SOURCE in the first place?

However, while while using _DEFAULT_SOURCE and `sig_t` would work with GNU/Linux and all BSDs, it would not work on e.g. System V, which happens to be the system that this version of 123 was originally built for! I strongly feel that if we're porting this old SVR4 Lotus 123 to other platforms we should observe POSIX and System V APIs first and foremost, and only reach for GNU and BSD APIs if it's necessary to do so. This might be a matter of personal opinion, but nevertheless a matter I feel we should have some alignment on.

Anyways, hope this explains why I choose to observe the  POSIX standard whenever possible, and [in this particular case](https://pubs.opengroup.org/onlinepubs/9699919799/functions/signal.html).